test/test_bdecode.cpp

```diff
@@ -206,6 +209,69 @@ TORRENT_TEST(dict_null_key)
 	TEST_EQUAL(d.int_value(), 1);
 }
 
+// soft error reported for dictionary with unordered keys
+TORRENT_TEST(dict_unordered_keys)
+{
+	char error_string[200];
+	{
+		char b[] = "d2:abi1e2:aai2ee";
+		bdecode_node e;
+		error_code ec;
+		int ret = bdecode(b, b + sizeof(b) - 1, e, ec);
+		TEST_EQUAL(ret, 0);
+		TEST_CHECK(e.has_soft_error(error_string));
```

Comment:
I'm not sure about returning an error message via a span. Is there a benefit over returning an error_code?